### PR TITLE
fix: add runtime field of jest config

### LINKED ISSUE #603: False positive for dev dependency listed in jest config
I am running knip on a TS application with the following knip & jest files:

knip.json
```json
{
  "$schema": "https://unpkg.com/knip@5/schema.json",
  "entry": ["src/server.ts", "src/commands/command.ts", "test/**/*.ts", "!test/mocks/**/*"],
  "paths": {
    "@/src/*": ["./src/*"],
    "@/test/*": ["./test/*"]
  },
  "project": ["src/**/*.ts", "test/**/*.ts"],
  "husky": {
    "config": [".husky/pre-commit"]
  },
  "prettier": {
    "config": [".prettierrc"]
  },
  "eslint": {
    "config": [".eslintrc"]
  },
  "jest": {
    "config": ["jest.config.js"]
  },
  "ignoreDependencies": ["@instana/collector", "@side/jest-runtime"]
}
```

jest.config.js
```js
const { pathsToModuleNameMapper } = require('ts-jest');
const { compilerOptions } = require('./tsconfig.json');

module.exports = {
  transform: {
    '^.+\\.ts$': '@swc/jest',
  },
  clearMocks: true,
  runtime: '@side/jest-runtime',
  testEnvironment: 'node',
  testPathIgnorePatterns: ['node_modules', 'dist'],
  verbose: false,
  collectCoverage: true,
  coverageReporters: ['clover', 'json', 'lcov', 'text', 'text-summary'],
  globalSetup: '<rootDir>/test/jest/globalSetup.ts',
  globalTeardown: '<rootDir>/test/jest/globalTeardown.ts',
  setupFiles: ['<rootDir>/test/jest/setupTests.ts'],
  moduleNameMapper: pathsToModuleNameMapper(compilerOptions.paths, { prefix: '<rootDir>/' }),
};
```

I am getting a false positive, claiming @side/jest-runtime is an unused dev dependency, but is is mentioned in the jest config under the `runtime` key

### PR DESCRIPTION
**Pull request by decadef20**
fix #603

### CODE REVIEW

**Comment by webpro** [comment]
Thanks! Could you please add something to the related fixture so we'll have some coverage?

File: package-lock.json
Comment (webpro) [inline package-lock.json:1]: This file shouldn't be added

Comment (decadef20) [reply package-lock.json:1]: updated

Comment (webpro) [reply package-lock.json:1]: Thanks! I also asked for https://github.com/webpro/knip/pull/639#issuecomment-2112074291 - any chance you could work that in as well?

Comment (decadef20) [reply package-lock.json:1]: I've missed this comment. Sure, I'll do it

Comment (webpro) [reply package-lock.json:1]: Maybe you've forgot to add the file? It doesn't seem to be in this branch yet.

File: packages/knip/fixtures/plugins/jest/jest.config.js [hunk 1]
```diff
@@ -26,4 +26,5 @@ module.exports = {
   ],
   testResultsProcessor: 'jest-phabricator',
   snapshotResolver: '<rootDir>/snapshotResolver.js',
+  runtime: '@side/jest-runtime',
 };
```
Comment (decadef20) [inline packages/knip/fixtures/plugins/jest/jest.config.js:30]: Updated

**Comment by webpro** [comment]
There seems to be a failing test, any chance you could look into it?

**Comment by webpro** [comment]
Is there still interest in wrapping up this pull request?

**Review by Freakyty — APPROVED** [review]

**Comment by webpro** [comment]
Closing due to inactivity.

**Pull request closed without merging**